### PR TITLE
Infra/custom datetime picker input

### PR DESCRIPTION
**Pull request by mendyEdri**
## Description
Support passing custom render input to `DateTimePicker`

## Changelog
Support passing custom render input to `DateTimePicker`

### CODE REVIEW

File: src/components/dateTimePicker/index.js [hunk 1]
```diff
@@ -93,7 +93,11 @@ class DateTimePicker extends Component {
     /**
      * style to apply to the iOS dialog header
      */
-    headerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number])
+    headerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
+    /**
+     * Render custom expandable input (requires expandable to be true)
+     */
+    renderExpandableInput: PropTypes.elementType
```
Comment (ethanshar) [inline src/components/dateTimePicker/index.js:100]: while for the TextField it made sense to call the prop `renderExpandleInput` I think in this case it's less relevant to the user what's expandable is. 
I think renderInput is enough

File: demo/src/screens/componentScreens/DateTimePickerScreen.js [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+          style={{backgroundColor: 'red', height: 40, width: '100%'}}
+          {...props}
+          onPress={() => {
+            toggle(true);
+          }}
+        />
+        <Text>{this.getCustomInputValue(value)}</Text>
+      </View>
+    );
+  };
```
Comment (ethanshar) [inline demo/src/screens/componentScreens/DateTimePickerScreen.js:28]: Maybe we can do it a little prettier 😬 
Like in the design of the ticket

Comment (mendyEdri) [reply demo/src/screens/componentScreens/DateTimePickerScreen.js:28]: Lol np

File: demo/src/screens/componentScreens/DateTimePickerScreen.js [hunk 3]
```diff
@@ -37,6 +62,17 @@ export default class DateTimePickerScreen extends Component {
...
+          Custom Input
+        </Text>
+        <DateTimePicker
+          containerStyle={{marginVertical: 20}}
+          title={'Date'}
+          placeholder={'Select a date'}
+          renderExpandableInput={this.renderCustomInput}
+          // timeFormat={'h:mm A'}
+          // value={new Date('2015-03-25T12:00:00-06:30')}
+        />
```
Comment (ethanshar) [inline demo/src/screens/componentScreens/DateTimePickerScreen.js:75]: On Android I can't see the example.
Change the root view to a ScrollView

**Review by ethanshar — APPROVED** [review]

**Pull request merged**